packages/api/ecs/async-operation/index.js

```diff
@@ -280,36 +235,20 @@ const updateAsyncOperation = async (params) => {
 
   const knex = await getKnexClient({ env });
 
-  try {
-    return await createRejectableTransaction(knex, async (trx) => {
-      await writeAsyncOperationToPostgres({
-        dbOutput,
-        env,
-        status,
-        trx,
-        updatedTime,
-        asyncOperationPgModel,
-      });
-      const result = await writeAsyncOperationToDynamoDb({
-        env,
-        status,
-        dbOutput,
-        updatedTime,
-        dynamoDbClient,
-      });
-      await writeAsyncOperationToEs({ env, status, dbOutput, updatedTime, esClient });
-      logger.info(`Successfully updated async operation to ${JSON.stringify(status)} with output: ${JSON.stringify(output)}`);
-      return result;
-    });
-  } catch (error) {
-    await revertAsyncOperationWriteToDynamoDb({
+  return await createRejectableTransaction(knex, async (trx) => {
+    const pgRecords = await writeAsyncOperationToPostgres({
+      dbOutput,
       env,
-      // Can we get away with hardcoding here instead of looking up
-      // the existing record?
-      status: 'RUNNING',
+      status,
+      trx,
+      updatedTime,
+      asyncOperationPgModel,
     });
-    throw error;
-  }
+    const result = translatePostgresAsyncOperationToApiAsyncOperation(pgRecords[0]);
+    await writeAsyncOperationToEs({ env, status, dbOutput, updatedTime, esClient });
+    logger.info(`Successfully updated async operation to ${JSON.stringify(status)} with output: ${JSON.stringify(output)}`);
+    return result;
+  });
 };
 
 /**
```

Comment:
JSDoc for the `runTask` function below still references DDB

Reply:
e7b1fa8e73